src/ui/public/kuery/functions/utils/get_fields_by_wildcard.js

```diff
@@ -0,0 +1,37 @@
+import _ from 'lodash';
+
+// Copied from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+function escapeRegExp(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+}
+
+/**
+ * Returns a list of fields matching the given pattern.
+ * @param {string} pattern - Field name with optional wildcards.
+ * @param indexPattern
+ * @return {Field[]} - a list of index pattern Fields
+ */
+export function getFieldsByWildcard(pattern, indexPattern) {
+  if (pattern.includes('*')) {
+    const replacedWildcardPattern = pattern.replace('\\*', '_KBN_WILDCARD_');
```

Comment:
This `_KBN_WILDCARD_` stuff is kinda gross, but I'm drawing a blank on how to do it more elegantly at the moment.

Reply:
Could do something like this:

```javascript
function getRegex(pattern) {
  let regex = '';
  let isEscapeSequence = false;
  for (const char of pattern) {
    if (isEscapeSequence) {
      regex += escapeRegExp(char);
      isEscapeSequence = false;
    } else if (char === '\\') {
      isEscapeSequence = true;
    } else if (char === '*') {
      regex += '.*';
    } else if (char === '?') {
      regex += '.';
    } else {
      regex += escapeRegExp(char);
    }
  }
  return `^${regex}$`;
}
```

Although then single escape characters are ignored, and I'm not sure if that's exactly how we want to handle them.